src/features/create-quiz/create-quiz-second-card/create-quiz-second-card.tsx

```diff
@@ -1,182 +1,289 @@
+import { useFormik } from "formik";
 import { Button } from "primereact/button";
 import { InputText } from "primereact/inputtext";
 import { Card } from "primereact/card";
 import { Dropdown } from "primereact/dropdown";
+import { classNames } from "primereact/utils";
 import { useEffect, useState } from "react";
 import { AnswerItem } from "./components/answer-item";
 
 import { observer } from "mobx-react";
 import { QuizStore } from "../store/CreateQuizStore";
 
-import {
-  addDoc,
-  collection,
-  doc,
-  serverTimestamp,
-  setDoc,
-} from "firebase/firestore";
+import { addDoc, collection, serverTimestamp } from "firebase/firestore";
 import { db } from "../../../config/firebase-config";
+import { toJS } from "mobx";
 
 export interface Question {
   answerList: AnswersList[];
-  questionName: string;
+  questionName?: string;
   questionType: string | null;
 }
+export interface InputErrors {
+  [key: string]: string;
+}
 
 export interface AnswersList {
-  isCorrectAnswer: boolean;
+  isCorrectAnswer?: boolean;
   answerName: string;
 }
 
 export const CreateQuizSecondCard: React.FC = observer(() => {
-  const [question, setQuestion] = useState<Question | null>(null);
-
-  const [questionName, setQuestionName] = useState<string>("");
+  const [radio, setRadio] = useState<number | null>(null);
+  const [answerList, setAnswerList] = useState<AnswersList[]>([
+    {
+      isCorrectAnswer: true,
+      answerName: "",
+    },
+  ]);
 
-  const [questionType, setQuestionType] = useState(null);
   const questionTypes = [
     { name: "Multiple answers" },
     { name: "Single answer" },
     { name: "Subjective" },
   ];
 
-  const onCityChange = (e: { value: any }) => {
-    setQuestionType(e.value);
+  const handleRadioChange = (id: number) => {
+    setRadio(id);
   };
 
-  const [answerList, setAnswerList] = useState<AnswersList[]>([
-    {
-      isCorrectAnswer: true,
-      answerName: "",
-    },
-  ]);
-
-  const handleInputChange = (value: any, index: number) => {
+  const handleAnswerInputChange = (value: any, index: number) => {
     const newList = answerList;
     newList[index].answerName = value;
-    setAnswerList(newList);
+    formik.values.answerList = [...newList];
+    setAnswerList([...newList]);
   };
 
   const addAnswer = () => {
     const newList = answerList;
     newList.push({ isCorrectAnswer: false, answerName: "" });
+    formik.values.answerList = [...newList];
     setAnswerList([...newList]);
   };
 
   const removeAnswer = (index: number) => {
-    let newList = answerList;
-    newList.splice(index, 1);
-    console.log(`Deleted item ${index}`);
+    let newList = answerList.filter((_, answerIndex) => answerIndex !== index);
```

Comment:
To remove an item from an array by its index you could also use the Javascript function `splice()`.